### ISSUE TITLE
Mesh loaded not correctly illuminated

### ISSUE THREAD

**Issue by MatheoDumont** [STORM-IRIT/Radium-Engine#807]
Hello !

So here is my issue:

I've loaded a mesh, that can be considered as a HeightMap, with a below and an above. And it is not illuminated correctly, the below side is always lit, even by inversing the normals or rotating the mesh.

So i thought that the entity has its own light rotating with it, but i can't find a way to change anything, and the LightManager doesn't have a derived class.

What do you think ? 

Here is the [mesh.zip](https://github.com/STORM-IRIT/Radium-Engine/files/6885442/sea_floor_decimated.zip).

The mesh how it should be oriented, with the up as the side to be illuminated: 
![image](https://user-images.githubusercontent.com/29271028/127155583-384fcf84-9c10-4e7f-bc72-290dd124b73d.png)

Thanks in advance !

**Comment by MathiasPaulin** [comment]
I check your file and the reasons why its normals are inverted when loading.
The problem comes from the configuration of assimp loader and the process steps (e.g. `aiProcess_FixInfacingNormals`) we request after loading a file.

The assimp documentation about the flag `aiProcess_FixInfacingNormals` explain that
``` 
This step tries to determine which meshes have normal vectors that are facing inwards and inverts them.
The algorithm is simple but effective: the bounding box of all vertices + their normals is compared against the volume of the bounding box of all vertices without their normals. 
This works well for most objects, problems might occur with planar surfaces. 
However, the step tries to filter such cases. The step inverts all in-facing normals. 
Generally it is recommended to enable this step, although the result is not always correct.
```
Your file is then a fail-case for this processing step.

As we already had some concerns with assimp processing steps, we will investigate how to ensure that files are loaded the right way and, possibly, allow the user to define which processing steps he wants on a per file basis in the future.

For now, you have 2 possibilities to make your object correctly loaded :
 1. In the file `src/IO/AssimpLoader/AssimpFileLoader.cpp`, at line 54, you can remove the flag `aiProcess_FixInfacingNormals` and your normals will be loaded the right way. (hopping that this will not introduce problems on other meshes). I recommend that you use this workaround and let us know if you have problems with other models.
 2. You can also invert your normals **after** loading the file but this will be less generic.
 
Thank you for the issue that remind us this assimp problem.

**Comment by MatheoDumont** [comment]
Thanks for the quick answer, it works and im not using any other meshs in this scene so that's okay.

**Comment by dlyr** [comment]
Seems fixed by #816
